Log a warning if model for default PU and no default datasource

gsmet opened the pull request:
@yrodiere could you check if this would have helped you?

I might make it an error in a few versions but would like to have a test run for at least one version.

gsmet committed:
Log a warning if model for default PU and no default datasource

yrodiere commented:
I'm not sure I would have noticed the warning above the huge stacktrace related to the entity manager not being found, but yeah that would be a good first step.

That being said, I just tried running my application against a version of Quarkus compiled with this patch, and I didn't get any warning. See here for the reproducer: https://github.com/yrodiere/hibernate-demos/tree/noentitymanager-reproducer/hibernate-search/hsearch-quarkus-kotlin ; just run ./mvnw clean test from that directory, after you built Quarkus locally from your branch.

gsmet committed:
Trigger an implied PU even if no JDBC datasources

This only if we don't have Hibernate Reactive around. This allows to
properly analyze the state and provide informing error messages to the
users.

gsmet commented:
I fixed the issue reported by Yoann.
